### DISCUSSION TITLE
classes/class_cubemap

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#153] [User-contributed notes]
# classes/class_cubemap

Inherits: ImageTextureLayered< TextureLayered< Texture< Resource< RefCounted< Object Six square textures representing the faces of a cube. Commonly used as a skybox. Description: A cubemap is made ...

https://docs.godotengine.org/en/stable/classes/class_cubemap.html

**Comment by aarqon** [comment]
May be worth noting that the order for a packed cubemap is Front (+X), Back (-X), Top (+Y), Bottom (-Y), Left (+Z), Right (-Z)? Was surprisingly hard to find this information...

**Reply by Calinou** [reply]
There's a PR adding this information. It was mistakenly left in draft state after the [corresponding godot-docs PR](https://github.com/godotengine/godot-docs/pull/9719) was merged, so I went and marked it as ready for review:

- https://github.com/godotengine/godot/pull/95396

**Reply by ArcticWolves** [reply]
Yes it is quite difficult to find this information, but you can refer to the 'Import Images' documentation for templates. https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_images.html

**Comment by ArcticWolves** [comment]
If you need the reverse effect of converting a Spherical 360 panorama into a Cubemap - Use this tool. https://jaxry.github.io/panorama-to-cubemap/

**Comment by admin** [comment]
*Originally posted by [**Rouli-M**](https://github.com/Rouli-M) on 2026-06-07 09:51:20 UTC*

> Beginner here. Converted the PanoramaSkyMaterial to a ShaderMaterial, edited the shader code, and I find myself with a Cubemap with an empty texture and no way to actually set the texture. Creating a Cubemap ressource separately and using that one in the ShaderMaterial doesnt gives an option to set the texture in the editor either. I find myself not knowing how to actually pass the cubemap texture, reading the shader didn't make it clearer as it takes a "samplerCube source_panorama : filter_linear, source_color, hint_default_black;" which is not a texture.
